core/src/main/kotlin/in/specmatic/core/git/GitOperations.kt

```diff
@@ -34,9 +34,20 @@ fun clone(workingDirectory: File, gitRepo: GitRepo): File {
     return cloneDirectory
 }
 
+fun checkout(workingDirectory: File, branchName: String) {
+    logger.log("Checking out branch: ${branchName}")
+    try {
+        SystemGit(workingDirectory.path).checkout(branchName);
+    } catch(exception: Exception) {
+        logger.debug("Could not checkout branch ${branchName}")
+        logger.debug(exception.localizedMessage ?: exception.message ?: "")
+        logger.debug(exception.stackTraceToString())
+    }
+}
+
 private fun clone(gitRepositoryURI: String, cloneDirectory: File) {
     try {
-        SystemGit(cloneDirectory.parent, "-", AzureAuthCredentials).shallowClone(gitRepositoryURI, cloneDirectory)
+        SystemGit(cloneDirectory.parent, "-", AzureAuthCredentials).clone(gitRepositoryURI, cloneDirectory)
```

Comment:
I am guessing it was necessary to move to clone instead of shallowClone for branch support?

Reply:
Yes, the shallow clone only clones the main branch & doesn't fetch the remote branches definitions